lib/internal/mime.js

```
@@ -0,0 +1,339 @@
+        MapIteratorPrototypeNext(entries)) {
+    if (done) break;
+    const [key, value] = keyValuePair;
+    const encoded = encode(value);
+    // Ensure they are separated
+    if (ret.length) ret += ';';
+    ret += `${key}=${encoded}`;
+  }
+  return ret;
+};
+
+class MIMEParams {
+  #data;
+  constructor(init = '') {
+    this.#data = parseParametersString(`${init}`);
+  }
+
+  delete(name) {
+    this.#data.delete(name);
+  }
+
+  get(name) {
+    const data = this.#data;
+    if (data.has(name)) {
+      return data.get(name);
+    }
+    return null;
+  }
+
+  has(name) {
+    return this.#data.has(name);
+  }
+
+  set(name, value) {
+    const data = this.#data;
+    NotHTTPTokenCodePoint.lastIndex = 0;
+    name = `${name}`;
+    value = `${value}`;
+    const invalidName = StringPrototypeSearch(name, NotHTTPTokenCodePoint) !== -1;
+    if (name.length === 0 || invalidName) {
+      throw new Error('Invalid MIME parameter name');
+    }
+    NotHTTPQuotedStringCodePoint.lastIndex = 0;
+    const invalidValue = StringPrototypeSearch(value, NotHTTPQuotedStringCodePoint) !== -1;
+    if (invalidValue) {
+      throw new Error('Invalid MIME parameter value');
+    }
+    data.set(name, value);
+  }
+
+  *entries() {
+    return yield* this.#data.entries();
+  }
+
+  *keys() {
+    return yield* this.#data.keys();
+  }
+
+  *values() {
+    return yield* this.#data.values();
+  }
+
+  *[SymbolIterator]() {
+    return yield* this.#data.entries();
+  }
+
+  toJSON() {
+    return MIMEParamsStringify(this);
+  }
+
+  toString() {
+    return MIMEParamsStringify(this);
+  }
+
+  // Used to act as a friendly class to stringifying stuff
+  // not meant to be exposed to users, could inject invalid values
+  static _data(o) {
+    return o.#data;
+  }
+}
+
+const MIMEParamsData = MIMEParams._data;
+delete MIMEParams._data;
+
+class MIME {
+  #type;
+  #subtype;
+  #parameters;
+  constructor(string) {
+    string = `${string}`;
```

Same here, doc says input is `string` but we stringify it anyway. I think we should throw on invalid input or change the doc saying that this can be `any` that will be stringified if that's the desired behavior. (this applies to other methods in here that do it this way)

---

see https://github.com/nodejs/node/pull/21128/files#r355488514 . I don't think our docs have clarity here and using strings is ideal to quick reading of the docs.